license-header-check/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'License Header Check'
+description: 'check license/copyright header'
+inputs:
+  included_file_patterns:
+    description: "included file pattern"
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get changed files
+      shell: bash
+      run: |
+        CURRENT_YEAR=$(date +%Y)
+        LICENSE_PATTERN="Copyright \(c\) .*?${CURRENT_YEAR}, NVIDIA CORPORATION."
+        
+        # Format file patterns
+        IFS="," read -r -a FILE_PATTERN_ARRAY <<< ${{ inputs.included_file_patterns }}
+        PATTERNS=""
+        for PATTERN in "${FILE_PATTERN_ARRAY[@]}"; do
+            PATTERNS+="$(echo "$PATTERN" | xargs -I{} echo "{}") "
```

Comment:
what does this for?

can we just put the patterns array in below FILES loop? and maybe simplify the find+grep combination if possible like
```
        for FILE in $FILES; do
            echo "Detected changed file: $FILE"
            for PATTERN in "${FILE_PATTERN_ARRAY[@]}"; do
                if [[ "$FILE" =~ $PATTERN ]]; then
```

Reply:
This is to trim the patterns in pattern array, I'll simplify the step here, maybe merge with `read` command. 

`[[ "$FILE" =~ $PATTERN ]]`  cannot work as expected with `wildcard`, I think `wildcard` as pattern could be clearer for uses compared with `regex`.

Reply:
Replaced `find` with `[[ $(basename $FILE) == $PATTERN ]]` as discussed in slack